command/agent/template/template.go

```diff
@@ -117,28 +108,26 @@ func (ts *Server) Run(ctx context.Context, incoming chan string, templates []*ct
 						Token: latestToken,
 					},
 				}
-				runnerConfig.Merge(&ctv)
-				runnerConfig.Finalize()
+				runnerConfig = runnerConfig.Merge(&ctv)
 				var runnerErr error
 				ts.runner, runnerErr = manager.NewRunner(runnerConfig, false)
 				if runnerErr != nil {
 					ts.logger.Error("template server failed with new Vault token", "error", runnerErr)
 					continue
-				} else {
-					go ts.runner.Start()
 				}
+				go ts.runner.Start()
 			}
 		case err := <-ts.runner.ErrCh:
 			ts.logger.Error("template server error", "error", err.Error())
-			close(ts.UnblockCh)
 			return
 		case <-ts.runner.TemplateRenderedCh():
```

Comment:
I think the tests are telling me that this is not the case, but does this stop on first render finish before exiting (if exit_after_auth is enabled)?

Reply:
Never mind, I took a look at the source and it seems that the channel receives something after all templates have been rendered. The channel simply indicates whether at least 1 of the templates got rendered, but nothing gets sent until all templates have been processed.